milli/src/update/delete_documents.rs

```diff
@@ -123,6 +131,21 @@ impl<'t, 'u, 'i> DeleteDocuments<'t, 'u, 'i> {
             }
         }
 
+        let mut fields_distribution = self.index.fields_distribution(self.wtxn)?;
+
+        for (field_id, count_diff) in fields_ids_distribution_diff {
+            let field_name = fields_ids_map.name(field_id).unwrap();
+            let count = fields_distribution.entry(field_name.to_string()).or_default();
+
+            if count == &count_diff {
+                fields_distribution.remove(field_name);
+            } else {
+                *count -= count_diff;
+            }
+        }
+
+        self.index.put_fields_distribution(self.wtxn, &fields_distribution)?;
```

Comment:
Could you please add a little comment explaining this algorithm, above it, please?

Reply:
```suggestion
        let mut fields_distribution = self.index.fields_distribution(self.wtxn)?;

        for (field_id, count_diff) in fields_ids_distribution_diff {
            let field_name = fields_ids_map.name(field_id).unwrap();
            if let Entry::Occupied(entry) = fields_distribution.entry(field_name.to_string()) {
                match entry.get().checked_sub(count_diff) {
                    Some(0) | None => entry.remove(),
                    Some(count) => entry.insert(count)
                };
            }
        }

        self.index.put_fields_distribution(self.wtxn, &fields_distribution)?;
```

What do you think about its implementation? It reduces the number of lookups in the database. Could you check that it is valid, please? On my side, it is :)

Reply:
LGTM and works. Didn't know about `checked_sub()`, nice way to check overflow safely.